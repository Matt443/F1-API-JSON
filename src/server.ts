import { getDriverData } from "./scraper/driver-data";
import { getTeamsData } from "./scraper/team-data";
import { getDriverStandings } from "./scraper/driver-standings";
import { getConstructorStandings } from './scraper/constructors-standings';
import { getWorldChampions } from './scraper/world-champions'

export {
    getDriverData,
    getTeamsData,
    getDriverStandings,
    getConstructorStandings,
    getWorldChampions
}