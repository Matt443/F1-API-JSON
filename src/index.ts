import { getRaceResults } from "./server";

async function test() {
    await getRaceResults();
    console.log();
}
test();
