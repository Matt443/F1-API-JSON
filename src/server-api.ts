import express from "express";
import cors from "cors";
import helmet from "helmet";
import dotenv from "dotenv";

import router from "./endpoints-api/routes-api";
import { getFullRaceResults, getRaceResults } from "./server";

dotenv.config();

const app = express();

app.use(cors());
app.use(helmet());

app.use("/", router);

app.listen(process.env.PORT || 4005, () => {
    console.log(`server is running on ${process.env.PORT || 4005}`);
}).on("error", (error) => {
    throw new Error(error.message);
});
