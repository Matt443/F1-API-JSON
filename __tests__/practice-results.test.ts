import { getFullPracticeResults } from "../src/server";

test("fastest laps", async () => {
    expect(await getFullPracticeResults(2024)).toMatchSnapshot();
});
