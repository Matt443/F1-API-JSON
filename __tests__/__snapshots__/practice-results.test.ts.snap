// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fastest laps 1`] = `
[
  {
    "code": "NOR",
    "laps": 23,
    "name": "Lando Norris",
    "number": 4,
    "position": "1",
    "team": "McLaren Mercedes",
    "time": "1:18.564",
  },
  {
    "code": "VER",
    "laps": 19,
    "name": "Max Verstappen",
    "number": 1,
    "position": "2",
    "team": "Red Bull Racing Honda RBPT",
    "time": "+0.018s",
  },
  {
    "code": "RUS",
    "laps": 21,
    "name": "George Russell",
    "number": 63,
    "position": "3",
    "team": "Mercedes",
    "time": "+0.033s",
  },
  {
    "code": "LEC",
    "laps": 22,
    "name": "Charles Leclerc",
    "number": 16,
    "position": "4",
    "team": "Ferrari",
    "time": "+0.035s",
  },
  {
    "code": "TSU",
    "laps": 26,
    "name": "Yuki Tsunoda",
    "number": 22,
    "position": "5",
    "team": "RB Honda RBPT",
    "time": "+0.057s",
  },
  {
    "code": "PER",
    "laps": 23,
    "name": "Sergio Perez",
    "number": 11,
    "position": "6",
    "team": "Red Bull Racing Honda RBPT",
    "time": "+0.078s",
  },
  {
    "code": "STR",
    "laps": 26,
    "name": "Lance Stroll",
    "number": 18,
    "position": "7",
    "team": "Aston Martin Aramco Mercedes",
    "time": "+0.103s",
  },
  {
    "code": "SAI",
    "laps": 23,
    "name": "Carlos Sainz",
    "number": 55,
    "position": "8",
    "team": "Ferrari",
    "time": "+0.122s",
  },
  {
    "code": "HAM",
    "laps": 20,
    "name": "Lewis Hamilton",
    "number": 44,
    "position": "9",
    "team": "Mercedes",
    "time": "+0.207s",
  },
  {
    "code": "PIA",
    "laps": 25,
    "name": "Oscar Piastri",
    "number": 81,
    "position": "10",
    "team": "McLaren Mercedes",
    "time": "+0.354s",
  },
  {
    "code": "RIC",
    "laps": 25,
    "name": "Daniel Ricciardo",
    "number": 3,
    "position": "11",
    "team": "RB Honda RBPT",
    "time": "+0.710s",
  },
  {
    "code": "ALB",
    "laps": 11,
    "name": "Alexander Albon",
    "number": 23,
    "position": "12",
    "team": "Williams Mercedes",
    "time": "+0.879s",
  },
  {
    "code": "MAG",
    "laps": 21,
    "name": "Kevin Magnussen",
    "number": 20,
    "position": "13",
    "team": "Haas Ferrari",
    "time": "+0.925s",
  },
  {
    "code": "SAR",
    "laps": 22,
    "name": "Logan Sargeant",
    "number": 2,
    "position": "14",
    "team": "Williams Mercedes",
    "time": "+0.955s",
  },
  {
    "code": "OCO",
    "laps": 25,
    "name": "Esteban Ocon",
    "number": 31,
    "position": "15",
    "team": "Alpine Renault",
    "time": "+0.997s",
  },
  {
    "code": "HUL",
    "laps": 21,
    "name": "Nico Hulkenberg",
    "number": 27,
    "position": "16",
    "team": "Haas Ferrari",
    "time": "+1.040s",
  },
  {
    "code": "GAS",
    "laps": 25,
    "name": "Pierre Gasly",
    "number": 10,
    "position": "17",
    "team": "Alpine Renault",
    "time": "+1.058s",
  },
  {
    "code": "ALO",
    "laps": 16,
    "name": "Fernando Alonso",
    "number": 14,
    "position": "18",
    "team": "Aston Martin Aramco Mercedes",
    "time": "+1.152s",
  },
  {
    "code": "ZHO",
    "laps": 23,
    "name": "Zhou Guanyu",
    "number": 24,
    "position": "19",
    "team": "Kick Sauber Ferrari",
    "time": "+1.425s",
  },
  {
    "code": "BOT",
    "laps": 21,
    "name": "Valtteri Bottas",
    "number": 77,
    "position": "20",
    "team": "Kick Sauber Ferrari",
    "time": "+1.450s",
  },
]
`;
